Microsoft.Toolkit.Uwp.UI.Controls/CameraPreview/CameraPreview.cs

```diff
@@ -0,0 +1,172 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.Helpers.CameraHelper;
+using Windows.Media.Capture.Frames;
+using Windows.Media.Core;
+using Windows.Media.Playback;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Camera Control to preview video. Can subscribe to video frames, software bitmap when they arrive.
+    /// </summary>
+    public partial class CameraPreview : Control, IDisposable
+    {
+        private CameraHelper _cameraHelper;
+        private MediaPlayer _mediaPlayer;
+        private MediaPlayerElement _mediaPlayerElementControl;
+        private Button _toggleFrameSourceGroup;
+        private int _selectedSourceIndex = 0;
+
+        /// <summary>
+        /// Gets Frame Source Groups available for Camera Media Capture.
+        /// </summary>
+        public IReadOnlyList<MediaFrameSourceGroup> FrameSourceGroups { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CameraPreview"/> class.
+        /// </summary>
+        public CameraPreview()
+        {
+            this.DefaultStyleKey = typeof(CameraPreview);
+        }
+
+        protected async override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            if (_toggleFrameSourceGroup != null)
+            {
+                _toggleFrameSourceGroup.Click -= ToggleFrameSourceGroup_ClickAsync;
+            }
+
+            _mediaPlayerElementControl = (MediaPlayerElement)GetTemplateChild("MediaPlayerElementControl");
+            _toggleFrameSourceGroup = (Button)GetTemplateChild("FrameSourceGroupButton");
+
+            if (_toggleFrameSourceGroup != null)
+            {
+                _toggleFrameSourceGroup.Click += ToggleFrameSourceGroup_ClickAsync;
+            }
+
+            await InitializeAsync();
+        }
+
+        private async Task InitializeAsync()
+        {
+            if (_cameraHelper == null)
+            {
+                _cameraHelper = new CameraHelper();
+                var result = await _cameraHelper.InitializeAndStartCaptureAsync();
+                if (result == CameraHelperResult.Success)
+                {
+                    // Subscribe to the frames as they arrive
+                    _cameraHelper.FrameArrived += CameraHelper_FrameArrived;
+                    FrameSourceGroups = _cameraHelper.FrameSourceGroups;
+                }
+                else
```

Comment:
Not sure what you mean?

Reply:
you are checking @line if 71: if (_cameraHelper == null) 
it will always be null since it is only assined in this method (InitializeAsync()) which is called only in OnApplyTemplate(). 
That said, I think it is valid to assume there could be a CameraPreview instantiated with an existing CameraHelper instance instead of a new one everytime, but your code so far does not allow to pass one as a parameter to the initialization of a CameraPreview.

Reply:
It is possible OnApplyTemplate to be called multiple times so that check is necessary in this context. We did discuss having CameraHelper act as a singleton potentially, so that might still be an option.